docs/tutorial/flow-nft-marketplace.md

```diff
@@ -0,0 +1,1804 @@
+
+After properly converting the object, we had to build a "payload" by calling `fcl.args` and pass an array of arguments. In this case, the metadata of type `[{key: string, value: string}]`.  To facilitate this, we used types from `fcl.types` library.
+
+Next, we fetch the corresponding `MintToken.cdc` raw code. This is a standard way of fetching raw text from another module.
+
+Now comes the meaty part of this function: Sending a transaction.
+
+```js
+
+const encoded = await fcl.send([
+  fcl.transaction(code),
+  fcl.payer(fcl.authz),
+  fcl.proposer(fcl.authz),
+  fcl.authorizations([fcl.authz]),
+  fcl.limit(999),
+  payload,
+]);
+
+```
+
+There is a few ways to send a transaction, but `fcl.send([...])` is the most explicit way.
+
+We pass the Cadence `code` to `fcl.transaction`, and any integer from 0 - 999 to `fcl.limit` for the gas fee limit we are happy with. The `payload` is the metadata we converted previously.
+
+The `payer`, `proposer`, and `authorizations` are a little complicate. While we won't dig deep into them, they accept a function known as *authorization function* that decides the account (and effectively the keys) used to authorize the transaction. (If you're interested in deep-diving, check out [Authorization Function][cdc-auth-function]). Here, `fcl` provided an `authz` default authorization function to makes signing with the emulator account easier.
+
+Now all that is left to do is to return to the main `mintToken` function to complete it:
+
+```js
+
+// This is a fallible function.
+async function mintToken(pet) {
+
+  // The metadata contains the attribute `url` which is an IFPS URL
+  // pointing to the data.json.
+  const { url } = await uploadToStorage(pet);
+
+  // We want to include the IPFS URL to the blockchain, so we can
+  // "unpack" the token data when we query it later. So we create
+  // a new object with all of the pet's attributes plus `url`.
+  const txId = await mintPet({ ...pet, url });
+  return txId;
+}
+
+// Don't forget to export the function.
+export default mintToken;
+
+```
+
+Our `mintToken` function is ready to work. We should return to `Form.js`,add a `handleSubmit` handler (right after `setAge` function), and pass to the `onSubmit` prop on the `<form>` element.
+
+```js
+
+// Form.js
+
+// On the top most of the module
+import mintToken from '../flow/transaction/MintToken.tx';
+
+const Form = () => {
+
+  // ... setAge function ...
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      await mintToken(pet);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  return (
+    <div style={style}>
+      <form onSubmit={handleSubmit}>
+
+        {/* ... other elements ... */}
+
+      </form>
+    </div>
+
+  );
+
+
+```
+
+And we wrap up the minting step! Here is the [full code][source] if you have lost your track. Test out the UI, select an image file, fill up the metadata on the form, and click the mint button.
```

Comment:
```suggestion
This wraps up the minting step! Here is the [full code][source] for a handy reference. Now you can test the UI, select an image file, fill up the metadata on the form, and click the mint button.
```
also, do you want to explicitly how to test the UI (ie load the js page into browser)?